### ISSUE TITLE
hasAccess() defaults to allow-all

### ISSUE THREAD

**Issue by dbohannon** [ksmithut/mongoose-role#2]
The hasAccess() method has a default allow-all configuration and will return true if called with no parameters. The roleHasAccess() method called by hasAccess() should be modified to return false if hasAccess() is called with no parameters, denying access unless explicitly granted. 

```
function roleHasAccess(role, accessLevels) {
    if (typeof accessLevels === 'undefined') { return true; } //should return false
    accessLevels = [].concat(accessLevels);
```

**Comment by ksmithut** [comment]
That's a good point. It's been a while since I originally developed this, and I want to say there was a reason for it, but perhaps I was mistaken. I'll look at this today. I know it's an easy fix, just want to make sure I figure out all the use cases as it will be a breaking change. Do you have an example of how you're using this but getting the unexpected behavior?

**Comment by dbohannon** [comment]
If the application tries to dynamically set the access level but passes an undefined value to hasAccess() then it will always return true.  The snippet below is a simple example where the accessLevel variable is supposed to be dynamically set, but is undefined when called by hasAccess().

```
var accessLevel;
app.get('/hack/getItem', function(req, res){
        for(i = 0; i < items.length; i++){
                if(items[i].name === req.query.name){
                        //this route calls the hasAccess() method with undefined var - will always return true
                        if(items[i].hasAccess(accessLevel))
                                res.send(items[i]);
                        else
                                res.sendStatus(401);
                        return;
                }
        }
        res.send("Item not found");
});
```

**Comment by ksmithut** [comment]
I guess that make sense. I'll do this, but it will be a major version bump since it's a breaking change. I'll have something published out later today.

**Comment by ksmithut** [comment]
@dbohannon Sorry for not getting to this sooner. It was literally a one word change and I still couldn't find the time to do this. I've published version 2.0.0 with your requested changes.

**Comment by dbohannon** [comment]
@ksmithut Looks good, thanks for taking the time to do this.
